tests/xbm/github_bug_170.c

```diff
@@ -0,0 +1,43 @@
+/*
+	Test writing of XBM images with a width that is not a multiple of 8
+
+	We create an image with a width of 11 pixels, and draw a circle on it.
+	To test that the padding is correctly applied, we write the image to disk
+	and assert that the number of bytes written matches our expectation.
+
+	See also <https://github.com/libgd/libgd/issues/170>.
+*/
+
+#include "gd.h"
+#include "gdtest.h"
+
+int main()
+{
+	gdImagePtr im;
+	int black;
+	FILE *outFile;
+	gdIOCtx *out;
+	long int length;
+
+	/* create the test image */
+	im = gdImageCreate(11, 11);
+	gdImageColorAllocate(im, 255, 255, 255);
+	black = gdImageColorAllocate(im, 0, 0, 0);
+	gdImageArc(im, 5, 5, 10, 10, 0, 360, black);
+
+	/* write the file to disk, note the file length and delete the file */
+	outFile = fopen("xbm/github_bug_170.xbm", "w");
+	if (outFile == NULL) return 1;
+	out = gdNewFileCtx(outFile);
+	if (out == NULL) return 1;
```

Comment:
please don't one-line statements.  that said, use the gd assert helpers here instead of checking/returning yourself.